src/lib/optimizer/strategy/chunk_pruning_rule.cpp

```diff
@@ -98,6 +104,8 @@ std::vector<PredicatePruningChain> find_predicate_pruning_chains_by_stored_table
       predicate_pruning_chains.emplace_back(current_predicate_pruning_chain);
       return LQPUpwardVisitation::DoNotVisitOutputs;
     }
+    // UnionNodes, among other node types, are never added to tracked_nodes because they cancel predicate chains.
```

Comment:
Where do the UnionNodes come from?

Reply:
This comment was meant as a hint. Technically, UnionNodes can get visited twice because they might be  next node of two separate predicate chains. However, because they lead to `predicate_pruning_chain_continues == false` we do not track them.

Reply:
(similar discussion as the comment below, let's try to figure that one out first)

Reply:
removed